### PR TITLE
WebView2: Add JS API docs for WebViewMessageEvent and cancelPromise

### PR DESCRIPTION
**Pull request by peiche-jessica**
Rendered webpages for review:
* **WebView2Script package** (root)
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/index?branch=pr-en-us-3110
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/)
* **WebViewMessageEvent interface** (new webpage)
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/webviewmessageevent?branch=pr-en-us-3110
* **WebViewEventMap interface** (new webpage)
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/webvieweventmap?branch=pr-en-us-3110
* **HostObjectsAsyncRoot class** > **cancelPromise(promise)** (new section in webpage)
   * https://review.learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsasyncroot?branch=pr-en-us-3110#webview2script-hostobjectsasyncroot-cancelpromise-member(1)
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/webview2/reference/javascript/hostobjectsasyncroot)

AB#49581036

### CODE REVIEW

File: microsoft-edge/webview2/reference/javascript/hostobjectsasyncroot.yml [hunk 1]
```diff
@@ -45,6 +45,23 @@ properties:
       return:
         type: '<xref uid="WebView2Script!HostObjectsSyncRoot:class" />'
 methods:
+  - name: cancelPromise(promise)
+    uid: 'WebView2Script!HostObjectsAsyncRoot#cancelPromise:member(1)'
+    package: WebView2Script!
+    fullName: cancelPromise(promise)
+    summary: Performs a best effort cancelation on promises for async method calls.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/hostobjectsasyncroot.yml:52]:
```suggestion
    summary: Performs a best-effort cancellation on promises for async method calls.
```

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+isPreview: false
+isDeprecated: false
+type: interface
+properties:
+  - name: additionalObjects
+    uid: 'WebView2Script!WebViewMessageEvent#additionalObjects:member'
+    package: WebView2Script!
+    fullName: additionalObjects
+    summary: >-
+      Additional DOM objects that are sent via PostJSONMessageWithAdditionalObjects. These objects can be of following
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:17]:
```suggestion
      Additional DOM objects that are sent via `PostJSONMessageWithAdditionalObjects`. These objects can be of the following
```
* the
* assuming backticks are supported here

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      types:
+      
+      DOM | Win32
+      ------- | ------------
+      [FileSystemFileHandle](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle) | ICoreWebView2FileSystemHandle (kind File)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:22]:
```suggestion
      [FileSystemFileHandle](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle) | ICoreWebView2FileSystemHandle (kind `File`)
```
if backticks are supported here & arguably appropriate, to differentiate English regular words vs. code key words

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      [FileSystemDirectoryHandle](https://developer.mozilla.org/docs/Web/API/FileSystemDirectoryHandle) ICoreWebView2FileSystemHandle (kind Directory)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:23]:
```suggestion
      [FileSystemDirectoryHandle](https://developer.mozilla.org/docs/Web/API/FileSystemDirectoryHandle) ICoreWebView2FileSystemHandle (kind `Directory`)
```
if backticks are supported here & arguably appropriate.
suggest some such formatting hints to differentiate English regular words vs. code key words.

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:22]: what's the rendered URL?  need to see rendered, to recommend formatting

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:23]: what's the rendered URL?  need to see rendered, to recommend formatting

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:22]: is this topic missing from the TOC?

Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:22]: Missing description: empty cell for webviewmessageevent in bottom of table at
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/?branch=pr-en-us-3110#classes

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:1]: garbled cell content R1C2 at
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/webviewmessageevent?branch=pr-en-us-3110#properties

image

File: microsoft-edge/webview2/reference/javascript/webvieweventmap.yml
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webvieweventmap.yml:1]: missing from TOC

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/webvieweventmap?branch=pr-en-us-3110

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:1]: missing from TOC

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/webviewmessageevent?branch=pr-en-us-3110

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+uid: 'WebView2Script!WebViewMessageEvent:class'
+package: WebView2Script!
+fullName: WebViewMessageEvent
+summary: ''
+remarks: ''
+isPreview: false
+isDeprecated: false
+type: interface
+properties:
+  - name: additionalObjects
```
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:12]: additionalObjects section has garbled table:  
image

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml
Comment (mikehoffms) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:1]: Properties section has garbled cell in table
image

rendered:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/webviewmessageevent?branch=pr-en-us-3110#properties

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    uid: 'WebView2Script!WebViewMessageEvent#additionalObjects:member'
+    package: WebView2Script!
+    fullName: additionalObjects
+    summary: |-
+      Additional DOM objects that are sent via PostJSONMessageWithAdditionalObjects. These objects can be of following
+      types:
+
+      DOM | Win32
```
Comment (yildirimcagri-msft) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:20]:
```suggestion
      | DOM | Win32 |

```

File: microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      ------- | ------------
```
Comment (yildirimcagri-msft) [inline microsoft-edge/webview2/reference/javascript/webviewmessageevent.yml:21]:
```suggestion
      | ------- | ------------ |

```

**Review by yildirimcagri-msft — APPROVED** [review]

**Review by mikehoffms — APPROVED** [review]
Approved PR per Writer/Editor review pass after fixes.

**Pull request merged**